File: 12. Reverse The Array - GFG/12.-reverse-the-array.py
#User function Template for python3

def reverseArray(arr,n):
    
    #code here
    for i in range(n//2):
        arr[i], arr[-i-1] = arr[-i-1], arr[i] 

#{ 
#  Driver Code Starts
#Initial Template for Python 3

#contributed by RavinderSinghPB
if __name__ == '__main__':
    tcs=int(input())
    
    for _ in range(tcs):
        n=int(input())
        arr=[int(x) for x in input().split()]
        
        reverseArray(arr,n)
        
        for e in arr:
            print(e,end=' ')
        print()
# } Driver Code Ends

File: Array/12. Reverse The Array - GFG/12.-reverse-the-array.cpp
// { Driver Code Starts
// Initial Template for C++

#include <iostream>
using namespace std;


 // } Driver Code Ends
// User function Template for C++

// Complete this function
void reverseArray(int arr[], int n) {
    // Your code here
    int low = 0;
    int high = n-1;
    
    while(low < high){
        int temp = arr[low];
        arr[low] = arr[high];
        arr[high] = temp;
        low++;
        high--;
    }
    
}

// { Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) cin >> arr[i];

        reverseArray(arr, n);

        for (int i = 0; i < n; i++) {
            cout << arr[i] << " ";
        }

        cout << endl;
    }
    return 0;
}  // } Driver Code Ends